src/herder/TransactionQueue.h

```diff
@@ -139,23 +139,24 @@ class TransactionQueue
 
 #ifdef BUILD_TESTS
     AddResult tryAdd(TransactionFrameBasePtr tx, bool submittedFromSelf,
-                     bool isLoadgenTx = false);
+                     uint32_t ledgerVersion, bool isLoadgenTx = false);
```

Comment:
It's kind of annoying that we need to pass the protocol version to every method. I think we can just store `ledgerVersion` in tx queue and update it when we handle a processed ledger (we do a bunch of upkeep every ledger anyways).

Reply:
FWIW this could be done as a followup; it would reduce the number of diffs quite a bit though.

Reply:
+1 I don't think we need to pipe ledgerVersion to TransactionQueue or store it as a member var -- just use mApp.getLedgerManager().getLastClosedLedgerHeader().header.ledgerVersion (there are already places in TransactionQueue that do that, e.g. in canAdd)